### DISCUSSION TITLE
2.3.10 Cortex problem with whois

### DISCUSSION THREAD

**Discussion by 0nnyx** [Security-Onion-Solutions/securityonion#2085] [Unsupported Versions]
Doing a test case in Hive and trying the CERTatPassiveDNS_2_0 on a domain or IP gives this error :
/Cortex-Analyzers/analyzers/CERTatPassiveDNS/whois.sh: line 2: whois: command not foundTraceback (most recent call last): File "/Cortex-Analyzers/analyzers/CERTatPassiveDNS/certat_passivedns.py", line 38, in <module> CERTatPassiveDNSAnalyzer().run() File "/Cortex-Analyzers/analyzers/CERTatPassiveDNS/certat_passivedns.py", line 18, in run self.report({'results': query(self.get_data(), int(self.limit))}) File "/Cortex-Analyzers/analyzers/CERTatPassiveDNS/whois_wrapper.py", line 58, in query return __process_results(__query(domain, limit)) File "/Cortex-Analyzers/analyzers/CERTatPassiveDNS/whois_wrapper.py", line 17, in __query s = check_output(['{}'.format(os.path.join(os.path.dirname(__file__), 'whois.sh')), '--limit {} {}'.format(limit, domain)], universal_newlines=True) File "/usr/lib/python3.6/subprocess.py", line 356, in check_output **kwargs).stdout File "/usr/lib/python3.6/subprocess.py", line 438, in run output=stdout, stderr=stderr)subprocess.CalledProcessError: Command '['/Cortex-Analyzers/analyzers/CERTatPassiveDNS/whois.sh', '--limit 100 google.com']' returned non-zero exit status 127.

Do I need to install anything extra to have it working ?

I can do a whois lookup from securityonion itself : 
[root@securityonion]# whois google.com
   Domain Name: GOOGLE.COM
   Registry Domain ID: 2138514_DOMAIN_COM-VRSN
   Registrar WHOIS Server: whois.markmonitor.com
   Registrar URL: http://www.markmonitor.com
   Updated Date: 2019-09-09T15:39:04Z
   Creation Date: 1997-09-15T04:00:00Z
   Registry Expiry Date: 2028-09-14T04:00:00Z
   Registrar: MarkMonitor Inc.
   Registrar IANA ID: 292

**Comment by weslambert** [comment]
I believe this is because the analyzer is calling the whois package on the backend, whois package is not included in the Docker image.  I'll make a note to include it in the install.
